dataflowAPI/src/ExpressionConversionVisitor.C

```diff
@@ -160,6 +161,14 @@ void ExpressionConversionVisitor::visit(RegisterAST *regast) {
     return;
 }
 
+void ExpressionConversionVisitor::visit(MultiRegisterAST *multiregast) {
+    // has no children
+    for (auto my_Reg : multiregast->getRegs() ){
+        visit(my_Reg.get());
+    }
```

Comment:
I don't know. If a compiler decides to push 2 registers on the stack, but pop 1 register off the stack, we would need to rewrite what is left on the stack? I feel like at the end of the day, MultiRegisterAST is just a concept for convenience, any underlying operation should be based on the architecture registers.

Reply:
This has nothing to do with code generated from a compiler; it has to do with the Instruction::Visitor usage.  The Visitor (misleadingly) visitor all the nodes in leftmost postfix order, and there is no mechanism to return values. So to use the visitor class uses a stack (the implementation C++ data structure may be a vector, list, dequeue or stack; but it really is a stack) to store data and computes values by popping the argument(s) if any, and pushing the results; on the stack a register has no arguments, a deref has one argument, a binary op has two, and all push a result. After the computation the stack should have one element and it is the result. If you push N results for a multiregister (which your code does in all cases, this give wrong results.

For example if the computation was to simplify 0+x to x, the expression `Immediate(0) + multiregister(a, b)`

```
Node               Stack after node
-----              ----------------
Immediate(0)       0
multireg(a, b)     0 register(a) register(b)
+                  0 BinaryOp(+, register(a), register(b))
```

This is an error since there is more than on element on the stack and the result is accidently right (you'd get the same result if the immediate were 99.  Now if the multireg is kept together as one element:

```
Node               Stack after node
-----              ----------------
Immediate(0)       0
multireg(a, b)     0 multireg(a, b)
+                  multireg(a, b)
```

which is the correct simplification and it would not simplify if the immediate were 99.